ministryofjustice/correspondence_tool_staff
# CT-2874 Experiment with alternative policy for FOI and Offender SAR

## Description
This is a follow up to the previous work on responder versus manager roles and who could see/edit offender sars. 

I'd done an implementation using scopes which wasn't ideal but the best I could see at the time. It turns out however, that the base policy simply delegates to the CorrespondenceType policy to decide who can see what at a basic level, and it turned out I'd not set that up sensibly last year. 

SO

This PR reverts some of the changes using scopes, and uses OffenderSAR policy to achieve the same goal but in a way that hopefully works better and handles cases like search which don't use the case_finder service but rely on the base policy returning sensible things. 

It's a pretty small change  - the main class affected other than offender SAR is the FOI policy which no longer allows anyone to view them (which was one of the things allowing FOI to show up for Branston responders e.g. in search)

This PR disables the "1 year ago" validation for Offender SAR cases, because we believe it will be necessary to be able to edit older cases and some operations that would otherwise be fine throw exceptions because the case fails date validation unexpectedly. 

## Self-review checklist


### Screenshots
#### BEFORE
David Attenborough can see Offender SAR cases when searching
![image](https://user-images.githubusercontent.com/1161161/84144881-e38d1100-aa50-11ea-8f81-dd6ae1498f53.png)

And Brian Rix can see FOI cases when searching
![image](https://user-images.githubusercontent.com/1161161/84144962-ff90b280-aa50-11ea-9070-20df8d42d6b8.png)

#### AFTER
David Attenborough now doesn't see Offender SAR cases... 
![image](https://user-images.githubusercontent.com/1161161/84145037-1a632700-aa51-11ea-9024-33350f3b7235.png)

...and Brian can't see FOI
![image](https://user-images.githubusercontent.com/1161161/84145078-2a7b0680-aa51-11ea-8778-e8db449149ff.png)

### Related JIRA tickets
https://dsdmoj.atlassian.net/browse/CT-2874

### Deployment
none

### Manual testing instructions
As Brian Rix, search for case number or subject that is FOI
As David Attenborough, search for case number or subject of Offender SAR
In both cases, should not see cases returned in search :) 
Also, David should not see Offender SAR in the open cases list and Brian Rix should not see FOI, SAR etc. (You may need to reseed or run data migrations in order to update your Brian Rix user and Branston team to responder, if you haven't already from the previous version of this)
